thorlcr/thorutil/thmem.cpp

```diff
@@ -646,7 +662,17 @@ void CThorExpandingRowArray::compact()
         }
     }
     numRows = freeFinger-rows;
-    // JCSMORE - this may be a good time to call IRowManager::compactRows
+    memsize_t numEmptiedPages = rowManager->compactRows(numRows, rows);
+
+#ifdef _DEBUG
+    ActPrintLog(&activity, "CThorExpandingRowArray::compact(): compactRows freed %"I64F"d pages", (unsigned __int64)numEmptiedPages);
+#endif
+
+    if (maxSpillCost)
```

Comment:
I think the row shrinking was only merged to master (https://github.com/hpcc-systems/HPCC-Platform/pull/6711).  That only shrinks huge pages - which will be done inline.